homeassistant/components/sabnzbd/sensor.py

```diff
@@ -1,39 +1,103 @@
 """Support for monitoring an SABnzbd NZB client."""
 from __future__ import annotations
 
-from homeassistant.components.sensor import SensorEntity
-from homeassistant.core import HomeAssistant
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-from . import (
-    DATA_SABNZBD,
-    SENSOR_TYPES,
-    SIGNAL_SABNZBD_UPDATED,
-    SabnzbdSensorEntityDescription,
+
+from . import DOMAIN, SIGNAL_SABNZBD_UPDATED, SabnzbdApiData
+from ...config_entries import ConfigEntry
+from ...const import DATA_GIGABYTES, DATA_MEGABYTES, DATA_RATE_MEGABYTES_PER_SECOND
+from ...core import HomeAssistant
+from ...helpers.entity_platform import AddEntitiesCallback
+from .const import KEY_API, KEY_NAME
+
+
+@dataclass
+class SabnzbdRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    key: str
+
+
+@dataclass
+class SabnzbdSensorEntityDescription(SensorEntityDescription, SabnzbdRequiredKeysMixin):
+    """Describes Sabnzbd sensor entity."""
+
+
+SENSOR_TYPES: tuple[SabnzbdSensorEntityDescription, ...] = (
+    SabnzbdSensorEntityDescription(
+        key="status",
+        name="Status",
+    ),
+    SabnzbdSensorEntityDescription(
+        key="kbpersec",
+        name="Speed",
+        native_unit_of_measurement=DATA_RATE_MEGABYTES_PER_SECOND,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="mb",
+        name="Queue",
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="mbleft",
+        name="Left",
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="diskspacetotal1",
+        name="Disk",
+        native_unit_of_measurement=DATA_GIGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="diskspace1",
+        name="Disk Free",
+        native_unit_of_measurement=DATA_GIGABYTES,
+    ),
+    SabnzbdSensorEntityDescription(
+        key="noofslots_total",
+        name="Queue Count",
```

Comment:
This is a total
```suggestion
        name="Queue Count",
        state_class=SensorStateClass.TOTAL,
```

Reply:
All other sensors also fit one of these 3 sensor classes. See the docs.

Reply:
added SensorStateClass to all sensors